Sources/Class-CacheAPI.php

```diff
@@ -0,0 +1,271 @@
+	 *
+	 * @access public
+	 * @param string $key The key to use, the prefix is applied to the key name.
+	 * @param mixed $value The data we wish to save.
+	 * @param string $ttl Overrides the default TTL.
+	 * @return bool Whether or not we could save this to the cache.
+	 */
+	public function putData($key, $value, $ttl = null);
+
+	/**
+	 * Specify custom settings that the cache API supports.
+	 *
+	 * @access public
+	 * @param array $config_vars Additional config_vars, see ManageSettings.php for usage.
+	 * @return void No return is needed.
+	 */
+	public function cacheSettings(array &$config_vars);
+
+	/**
+	 * Closes connections to the cache method.
+	 *
+	 * @access public
+	 * @return bool Whether or not we could close connections.
+	 */
+	public function quit();
+
+	/**
+	 * Gets the latest version of SMF this is compatible with.
+	 *
+	 * @access public
+	 * @return string the value of $key.
+	 */
+	public function getCompatibleVersion();
+
+	/**
+	 * Gets the min version that we support.
+	 *
+	 * @access public
+	 * @return string the value of $key.
+	 */
+	public function getMiniumnVersion();
+}
+
+/**
+ * Class cache_api
+ */
+abstract class cache_api implements cache_api_interface
+{
+	/**
+	 * @var string The last version of SMF that this was tested on. Helps protect against API changes.
+	 */
+	protected $version_compatible = 'SMF 2.1 Beta 3';
+
+	/**
+	 * @var string The minimum SMF version that this will work with
+	 */
+	protected $min_smf_version = 'SMF 2.1 Beta 3';
+
+	/**
+	 * @var string The prefix for all keys.
+	 */
+	protected $prefix = '';
+
+	/**
+	 * @var string The default TTL.
+	 */
+	protected $ttl = 120;
+
+	/**
+	 * Does basic setup of a cache method when we create the object but before we call connect.
+	 *
+	 * @access public
+	 * @return void No return is needed.
+	 */
+	public function __construct()
+	{
+		global $boardurl, $cachedir;
+
+		// Set our default prefix. setPrefix can still change this.
+		$this->setPrefix('');
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function isSupported($test = false)
+	{
+		global $cache_enable;
+
+		if ($test)
+			return true;
+		return !empty($cache_enable);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function connect()
+	{
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function setPrefix($prefix = '')
+	{
+		global $boardurl, $cachedir;
+
+		// Set the default if no prefix was specified.
+		if (empty($prefix))
+			$this->prefix = md5($boardurl . filemtime($cachedir . '/' . 'index.php')) . '-SMF-';
+		else
+			$this->prefix = $prefix;
+
+		return true;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getPrefix()
+	{
+		return $this->prefix;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function setDefaultTTL($ttl = 120)
+	{
+		$this->ttl = $ttl;
+
+		return true;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getDefaultTTL()
+	{
+		return $this->ttl;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getData($key, $ttl = null)
```

Comment:
These empty stubs should be removed since they all should already have been implemented in the extending objects.